multiversx/mx-chain-go
# Fix missing role + Use setSpecialRole function in tests

## Reasoning behind the pull request
- Fix missing `ESDTRoleNFTUpdate` role
- use `setSpecialRole` in chain simulator tests instead of setting role directly
- remove irrelevant integration test

## Testing procedure
- Standard system test
- Custom scenario: check that `ESDTMetaDataUpdate` with role `ESDTRoleNFTUpdate` is working properly
     - issue token (NFT, SFT, metaESDT) --> set special role `ESDTRoleNFTUpdate` --> update token metadata with `ESDTMetaDataUpdate` function


## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
